credential/base/src/main/java/org/wildfly/security/password/interfaces/RawScramDigestPassword.java

```diff
@@ -54,14 +55,16 @@ public RawScramDigestPassword clone() {
     }
 
     public int hashCode() {
-        return multiHashOrdered(multiHashOrdered(multiHashOrdered(Arrays.hashCode(digest), Arrays.hashCode(salt)), iterationCount), getAlgorithm().hashCode());
+        return multiHashOrdered(multiHashOrdered(multiHashOrdered(Arrays.hashCode(digest),
+                Arrays.hashCode(salt)), iterationCount), getAlgorithm().hashCode());
     }
 
     public boolean equals(final Object obj) {
         if (! (obj instanceof RawScramDigestPassword)) {
             return false;
         }
         RawScramDigestPassword other = (RawScramDigestPassword) obj;
-        return iterationCount == other.iterationCount && getAlgorithm().equals(other.getAlgorithm()) && Arrays.equals(digest, other.digest) && Arrays.equals(salt, other.salt);
+        return iterationCount == other.iterationCount && getAlgorithm().equals(other.getAlgorithm())
+                && MessageDigest.isEqual(digest, other.digest) && MessageDigest.isEqual(salt, other.salt);
```

Comment:
Just a very small comment, we can continue to use Arrays.equals for the salt comparison.

Reply:
👍